modules/core/src/main/scala/lucuma/odb/api/model/ConstraintSetModel.scala

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.api.model
+
+import cats._
+import cats.data.State
+import cats.syntax.all._
+import clue.data.Input
+import eu.timepit.refined.types.string.NonEmptyString
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import lucuma.core.enum._
+import lucuma.core.model.{ ConstraintSet, Program }
+import lucuma.core.optics.syntax.lens._
+import lucuma.odb.api.model.Existence._
+import lucuma.odb.api.model.syntax.input._
+import monocle.{ Fold, Lens, Optional }
+import monocle.macros.GenLens
+
+final case class ConstraintSetModel(
+  id:              ConstraintSet.Id,
+  existence:       Existence,
+  programId:       Program.Id,
+  name:            NonEmptyString,
+  imageQuality:    ImageQuality,
+  cloudExtinction: CloudExtinction,
+  skyBackground:   SkyBackground,
+  waterVapor:      WaterVapor,
+  elevationRange:  ElevationRangeModel
+)
+
+object ConstraintSetModel extends ConstraintSetModelOptics {
+
+  implicit val TopLevelConstraintSet: TopLevelModel[ConstraintSet.Id, ConstraintSetModel] =
+    TopLevelModel.instance(_.id, ConstraintSetModel.existence)
+
+  implicit val EqConstraintSet: Eq[ConstraintSetModel] = Eq.fromUniversalEquals
+
+  final case class Create(
+    constraintSetId: Option[ConstraintSet.Id],
+    programId:       Program.Id,
+    name:            String,
+    imageQuality:    ImageQuality,
+    cloudExtinction: CloudExtinction,
+    skyBackground:   SkyBackground,
+    waterVapor:      WaterVapor,
+    elevationRange:  ElevationRangeModel.Create
```

Comment:
Should a user be required to specify all of these fields in order to create a constraints set?  Should any of them be optional?  It seems like we could default most of the values?

Reply:
I went back and forth on that. In the UI, they will essentially be forced to enter them because they will be defaulted in the dropdowns. Well, except for the elevation range which will be decimal inputs, but those would also have "defaults". But, for users of the API, I guess maybe we should provide defaults for them. I'll ask Andy what those should be.

Reply:
Right, for example `waterVapor` won't matter as I understand it for optical observations.  Ideally you would only have to specify values you care about.

Reply:
Andy says it's fine to require all values in the API. I told him we can switch to allowing defaults later if he changes his mind.

Reply:
We can certainly start with this.  I try to keep in mind the poor schlub sitting in front of a Playground UI typing in the stuff needed to make an observation.  There are a lot of parameters here that have to be specified even if most of them would be defaultable by making these `Option`s and then getting the least restrictive value when not specified.

Reply:
Agreed. I can run it by Andy again before the epic is over. Maybe part of it is wanting to avoid confusion over expectations. In the UI the defaults won't be the least restrictive values.